python/src/space/core/runners.py

```diff
@@ -256,3 +276,29 @@ def delete(self, filter_: pc.Expression) -> Optional[rt.Patch]:
                          self._storage.data_files(filter_), filter_,
                          self._file_options)
     return op.delete()
+
+  @StorageMixin.reload
+  def add_tag(self, tag:str, snapshot_id: Optional[int] = None) -> JobResult:
+    """Add Tag to a snapshot."""
+    try:
+      self._storage.add_tag(tag, snapshot_id)
```

Comment:
Can we extract common part into a method, maybe called `wrap_errors`? It can accept a lambda that runs `self._storage.add_tag(tag, snapshot_id)` or `self._storage.remove_tag(tag)`.

Reply:
Done